gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/integration/IntegrationAgentImpl.java

```diff
@@ -60,8 +61,8 @@ public class IntegrationAgentImpl implements IntegrationAgent {
 
     private final ExchangeController exchangeController;
 
-    public IntegrationAgentImpl(@Qualifier("integrationExchangeController") ExchangeController exchangeController) {
-        this.exchangeController = exchangeController;
+    public IntegrationAgentImpl(@Qualifier("integrationExchangeController") Optional<ExchangeController> exchangeController) {
+        this.exchangeController = exchangeController.orElse(null);
```

Comment:
Won't you have NPE when trying to use this service and send commands via any send****Command methods?

Reply:
Yes but it can only happen if someone calls Integration endpoints directly (like the discovery or ingest) while the feature is enabled. As I wanted to spend some time on test I thought that was not worthing to take more time on this 🤷

Reply:
As the UI is active, don’t you think it will happen sooner or later?

Reply:
I've implement something to prevent NPE